### PR TITLE
Fix issues with metrics publishing DI registration

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Pulling ParticipantPublishUploadMetric DI registration out of ServiceCollectionExtensions.cs in Piipan.Participants.Core.

## Why?

Fix for Nac-574. The DI registration for ParticipantPublishUploadMetric is breaking the Orchestrator Function because required Env variables aren't set.

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantPublishUploadMetric.cs [hunk 1]
```diff
@@ -13,16 +14,35 @@ namespace Piipan.Participants.Core.Services
...
-                    default
-                );
+                string eventGridUriString = Environment.GetEnvironmentVariable("EventGridEndPoint");
+                string eventGridKeyString = Environment.GetEnvironmentVariable("EventGridKeyString");
 
+                if (eventGridUriString != null && eventGridKeyString != null)
+                {
+                    _client = new EventGridPublisherClient(
+                        new Uri(eventGridUriString),
+                        new AzureKeyCredential(eventGridKeyString),
```
Comment (ccaprio97) [inline participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantPublishUploadMetric.cs:37]: what happens if this fails?

File: participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantService.cs [hunk 2]
```diff
@@ -106,9 +104,9 @@ public async Task AddParticipants(IEnumerable<IParticipant> participants, string
 
                      _logger.LogInformation(uploadDBStatus);
 
-                    await _participantPublishUploadMetric.PublishUploadMetric(participanUploadMetrics);    
-
                     scope.Complete();
+
+                    await _participantPublishUploadMetric.PublishUploadMetric(participanUploadMetrics);
```
Comment (ccaprio97) [inline participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantService.cs:109]: nitpick: do we care about adding the t in participantUploadMetrics?

Comment (bgfoshay) [reply participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantService.cs:109]: I'll fix it.

File: participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantPublishUploadMetric.cs (hunk 1 above)
Comment (bgfoshay) [reply participants/src/Piipan.Participants/Piipan.Participants.Core/Services/ParticipantPublishUploadMetric.cs]: Good point. I've added in some error handling.

**Review by ccaprio97 — APPROVED** [review]
LGTM

**Pull request merged**